docs/extension-system/viewer-editor-apps.md

```diff
@@ -0,0 +1,135 @@
+---
+title: 'Viewer and editor apps'
+date: 2024-03-12T00:00:00+00:00
+weight: 60
+geekdocRepo: https://github.com/owncloud/web
+geekdocEditPath: edit/master/docs/extension-system
+geekdocFilePath: viewer-editor-apps.md
+geekdocCollapseSection: true
+---
+
+{{< toc >}}
+
+## Viewer and editor apps
+
+The ownCloud Web allows developers to implement apps for viewing and editing specific file types. For instance, the built-in preview app serves as the default application for opening media files like images, videos, or audio.
+
+This section will guide you through the process of implementing such an app within ownCloud Web.
+
+### Basic app structure
+
+An app is essentially a distinct package that must be specified as an external application in the Web configuration.
+
+The structure of an app is quite simple and straightforward. Consider, for example, the [draw.io app](https://github.com/owncloud/web/tree/master/packages/web-app-draw-io). It consists of a `package.json` file, a `src` directory containing all the source code, and a `l10n` directory for translations. Optionally, you may also include a `tests` directory if your application requires testing.
+
+### App setup
+
+Inside the `src` folder you will need a `index.ts` file that sets up the app so it can be registered by the Web runtime. It follows the basic structure as described in [the apps section]({{< ref "_index.md#apps" >}}), so it may look like this:
```

Comment:
```suggestion
Inside the `src` folder you will need an `index.ts` file that sets up the app so it can be registered by the Web runtime. It follows the basic structure as described in [the apps section]({{< ref "_index.md#apps" >}}), so it may look like this:
```